test/e2e/far_e2e_test.go

```diff
@@ -0,0 +1,185 @@
+package e2e
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	corev1 "k8s.io/api/core/v1"
+	apiErrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/medik8s/fence-agents-remediation/api/v1alpha1"
+	farController "github.com/medik8s/fence-agents-remediation/controllers"
+	"github.com/medik8s/fence-agents-remediation/pkg/cli"
+	farUtils "github.com/medik8s/fence-agents-remediation/test/e2e/utils"
+)
+
+const (
+	testNamespace  = "fence-agents-remediation"
+	fenceAgentIPMI = "fence_ipmilan"
+
+	// eventually parameters
+	timeout       = 2 * time.Minute
+	timeoutReboot = 3 * time.Minute
+	pollInterval  = 10 * time.Second
+)
+
+var _ = Describe("FAR E2e", func() {
+	testShareParam := map[v1alpha1.ParameterName]string{
+		"--username": "admin",
+		"--password": "password",
+		"--action":   "reboot",
+		"--ip":       "192.168.111.1",
+		"--lanplus":  "",
+	}
+	testNodeParam := map[v1alpha1.ParameterName]map[v1alpha1.NodeName]string{
+		"--ipport": {
+			"master-0": "6230",
+			"master-1": "6231",
+			"master-2": "6232",
+			"worker-0": "6233",
+			"worker-1": "6234",
+			"worker-2": "6235",
+		},
+	} // get ports
+	Context("fence agent - fence_ipmilan", func() {
+		var (
+			far                *v1alpha1.FenceAgentsRemediation
+			nodeBootTimeBefore time.Time
+			errBoot            error
+			testNodeName       string
+		)
+		nodes := &corev1.NodeList{}
+		BeforeEach(func() {
+			Expect(k8sClient.List(context.Background(), nodes, &client.ListOptions{})).ToNot(HaveOccurred())
+			if len(nodes.Items) <= 1 {
+				Fail("there is one or less available nodes in the cluster")
+			}
+			//TODO: Randomize the node selection
+			// Use FA on the first node - master-0
+			nodeObj := nodes.Items[0]
+			testNodeName = nodeObj.Name
+			log.Info("Testing Node", "Node name", testNodeName)
+
+			// save the node's boot time prior to the fence agent call
+			nodeBootTimeBefore, errBoot = getNodeBootTime(testNodeName)
+			if errBoot != nil {
+				Fail("Can't get boot time of the node")
```

Comment:
nit, a bit nicer: `Expect(errBoot).ToNot(HaveOccurred(), "failed to get boot time of the node")`